crates/starknet/src/starknet/events.rs

```diff
@@ -0,0 +1,425 @@
+        assert!(check_if_filter_applies_for_event(&address, &Some(keys_filter), &event));
+
+        // filter with incorrect address and correct filter keys
+        let address = Some(ContractAddress::new(Felt::from(0)).unwrap().try_into().unwrap());
+        let keys_filter = vec![vec![Felt252::from(2), Felt252::from(3)]];
+        assert!(!check_if_filter_applies_for_event(&address, &Some(keys_filter), &event));
+
+        // filter with correct address and incorrect filter keys
+        let address = Some(dummy_contract_address().try_into().unwrap());
+        let keys_filter = vec![vec![Felt252::from(1), Felt252::from(3)]];
+        assert!(!check_if_filter_applies_for_event(&address, &Some(keys_filter), &event));
+
+        // filter with incorrect address and incorrect filter keys
+        let address = Some(ContractAddress::new(Felt::from(0)).unwrap().try_into().unwrap());
+        let keys_filter = vec![vec![Felt252::from(1), Felt252::from(3)]];
+        assert!(!check_if_filter_applies_for_event(&address, &Some(keys_filter), &event));
+    }
+
+    #[test]
+    fn pagination_for_latest_block_that_has_1_transaction_with_5_events() {
+        let starknet = setup();
+
+        // no pagination to the latest block events
+        let (events, has_more) = get_events(
+            &starknet,
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            None,
+            None,
+            0,
+            None,
+        )
+        .unwrap();
+
+        assert_eq!(events.len(), 5);
+        assert!(!has_more);
+
+        // limit the result to only events, should be left 2 more
+        let (events, has_more) = get_events(
+            &starknet,
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            None,
+            None,
+            0,
+            Some(3),
+        )
+        .unwrap();
+
+        assert_eq!(events.len(), 3);
+        assert!(has_more);
+
+        // skip 3 events and return maximum 3, but the result should be 2, because the total amount
+        // of events for the latest block is 5
+        let (events, has_more) = get_events(
+            &starknet,
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            Some(BlockId::Tag(starknet_rs_core::types::BlockTag::Latest)),
+            None,
+            None,
+            3,
+            Some(3),
+        )
+        .unwrap();
+
+        assert_eq!(events.len(), 2);
+        assert!(!has_more);
+    }
+
+    #[test]
+    fn pagination_for_events_of_multiple_blocks() {
+        let starknet = setup();
+
+        // returns all events from all blocks
+        let (events, has_more) = get_events(&starknet, None, None, None, None, 0, None).unwrap();
+
+        assert_eq!(events.len(), 15);
```

Comment:
In many places in tests, there are assertions with .len(). For example here, we are getting 15 events but actually, we don't check if the events are correct. The wrong implementation of adaptations to the next version can return just a vector with 15 empty record and the test will still pass.